### ISSUE TITLE
Rule: Warn on unwrapped IIFEs

### ISSUE THREAD

**Issue by nzakas** [eslint/eslint#2]
Should warn whenever we see this pattern:

``` js
var foo = function() {

}();
```

There should be parentheses around the function.

Rule name should be `immed`
